addons/mail/static/src/xml/composer.xml

```diff
@@ -11,7 +11,14 @@
                 <div class="o_composer_input">
                     <textarea class="o_input o_composer_text_field" tabindex="2" placeholder="Write something..."/>
                     <div class="o_chatter_composer_tools">
-                        <button tabindex="4" class="btn btn-sm btn-icon fa fa-smile-o o_composer_button_emoji" type="button" data-toggle="popover"/>
+                        <!--
+                            Safari browser not supported focus event on buttons.
+                            So replacing <button> to the <a> tag works for bootstrap popover.
+
+                            https://github.com/twbs/bootstrap/issues/15935
+                            https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+                         -->
+                        <a tabindex="4" class="btn btn-sm btn-icon fa fa-smile-o o_composer_button_emoji" type="button" data-toggle="popover"/>
```

Comment:
@asa-odoo I do not understand what the problem was. I wonder if this patch is still relevant.

Reply:
Hello @adr-odoo 
No, this patch is now not relevant as we have changed the emoji toggle button behavior ( https://github.com/odoo/odoo/blob/10.0/addons/mail/static/src/js/composer.js#L441-L456 [popover] ]to https://github.com/odoo/odoo/blob/11.0/addons/mail/static/src/js/composer.js#L790-L807 [emoji panel]).